### ISSUE TITLE
Changing a block should add "edited" action to all versions affected

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#17036]
### Describe the problem and steps to reproduce it:

When we edit a block, for example to decrease the block range, the reviewer tools page shows misleading information on the formerly blocked versions as those do not have an action that the block was updated.

STR:
1. Submit a new add-on, and a submit an update, so you have two versions.
2. Block all versions.
3. On the reviewer tools page, both versions have a _Block added_ action.
4. Edit the block to only cover the first version.

### What happened?

On the reviewer tools page, the first version has a _Block edited_ action. the second version does not.

### What did you expect to happen?

On the reviewer tools page, both versions have a _Block edited_ action, as they are both affected by this block change.

### Anything else we should know?

Logically, the first version is not affected by the change of the block, since it was blocked before and still is. For the sake of consistency and probably easy if implementation, let's consider such versions as being _affected_ and thus show the _Block edited_ action for them.

Prioritizing as P2 as this can lead to confusion when add-ons have both listed and unlisted versions and blocks or block changes affect both channels.

**Comment by eviljeff** [comment]
The thinking at the time of the implementation was that version status would show the current blocklisting state of that version, and the review log shows what happened.  I.e. the reviewer would look primarily at the status of the version because that's what the current state is.  (This is consistent with other actions like approval status of a version).  

But if the lack of the log entry for edits that unblock a version is misleading we can add that too.  

Does this only apply to block edits going forward, or is work required to backfill existing block edits?  (Will be non-trivial to reconcile the Block and BlocklistSubmission records to work out which versions were unblocked, but -probably- not impossible)

**Comment by wagnerand** [comment]
> The thinking at the time of the implementation was that version status would show the current blocklisting state of that version, and the review log shows what happened. I.e. the reviewer would look primarily at the status of the version because that's what the current state is. (This is consistent with other actions like approval status of a version).

Thank you for providing context here. That makes sense and I also think, like blocking a version, unblocking a version is something that "happened" to the version to show in the review log.
 
> Does this only apply to block edits going forward, or is work required to backfill existing block edits? (Will be non-trivial to reconcile the Block and BlocklistSubmission records to work out which versions were unblocked, but -probably- not impossible)

Backfill would be rather nice to have. We can probably skip it as I don't think there would be too many. Would there be an easy-ish way to identify the ones that need to be backfilled, even manually through an sql query? I could run that and then just add a comment on the review page.

**Comment by eviljeff** [comment]
> Backfill would be rather nice to have. We can probably skip it as I don't think there would be too many. Would there be an easy-ish way to identify the ones that need to be backfilled, even manually through an sql query? I could run that and then just add a comment on the review page.

Not that I can see - we'd either need to iterate though all the block submissions (and kinto imports) to work out if any of them reduced the spread of versions affected; or try to work backwards from the current Block instances and what versions are currently blocked vs. versions that have `BLOCKLIST_BLOCK_ADDED` - if the version isn't blocked _now_ but has an `BLOCKLIST_BLOCK_ADDED` log entry then it's been edited.  

The former would be more accurate but very difficult to do, given how the submission objects are structured (they we originally only meant to be kept for the lifecycle of the approval); the latter would miss edge cases where the block range was edited repeatedly, and we wouldn't have the correct date for the entry (we could make a guess though).

**Comment by eviljeff** [comment]
#17081 will add the activity log; I didn't tackle backfilling.  If backfilling (via code) is still a requirement please log a follow-up issue.

**Comment by AlexandraMoga** [comment]
This issue is verified fixed on -dev.

The block edit action is now displayed in Reviewer tools Version history for versions that have been excluded from a block range:

https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/248690015d2f4394ac15
![image](https://user-images.githubusercontent.com/31961530/117459389-03280480-af54-11eb-896d-d8bd69c663bd.png)
